[Bug]: AutoFactory fails for nullable relational fields when related_model is still a string reference

### Versions

GeneralManager 0.39.1 • Python 3.14.3 • Django 6.0.4 • pytest 9.0.3 • pytest-django 4.12.0

### OS

Linux dev container

### Environment

Docker / Container

### What happened?

Calling a manager factory without explicitly passing values fails when the model contains nullable/blank relational fields whose `related_model` is still a string reference.

Example:

```python
self.project = Project.Factory.create()
```

This raises:

```text
general_manager.factory.factories.InvalidRelatedModelTypeError:
Related model for project_number must be a Django model class, got 'financials.AccountNumber'.
```

The field is nullable and has `default=None`, so I would expect the factory either to use `None` or to resolve the related model correctly before trying to create a related instance.

## What did you expect to happen instead?

`Project.Factory.create()` should create a valid `Project` instance without requiring explicit parameters for nullable relational fields.

This expectation is based on the documented factory support behavior:
https://timkleindick.github.io/general_manager/concepts/interfaces/existing_model_interface/?h=fact#factory-support

In this case, passing `project_number=None` explicitly works around the failure, but the factory should not require that for a nullable field with `default=None`.

### Steps to Reproduce / Minimal Repro

1. Define a GeneralManager with a `DatabaseInterface`.
2. Add a nullable relation using a Django string model reference.
3. Call `Manager.Factory.create()` without explicitly passing that field.

Minimal reproducible example:

```python
from typing import ClassVar

from django.db import models
from general_manager.interface import DatabaseInterface
from general_manager.manager import GeneralManager


class Project(GeneralManager):
    class Interface(DatabaseInterface):
        project_number = models.OneToOneField(
            "financials.AccountNumber",
            on_delete=models.CASCADE,
            null=True,
            blank=True,
            default=None,
            related_name="project_number",
        )

    class Factory:
        pass


def test_project_factory_create_without_explicit_nullable_relation():
    Project.Factory.create()
```

A concrete failing example from our project:

```python
self.project = Project.Factory.create()
```

Workaround:

```python
self.project = Project.Factory.create(
    project_number=None,
)
```

### Error output / Traceback

```shell
Running pytest with args: ['-p', 'vscode_pytest', '--rootdir=/workspaces/knowledge_hub/knowledgehub_generalmanager', '/workspaces/knowledge_hub/knowledgehub_generalmanager/backend/apps/project_management/tests/integration/test_change_request_graphql_capabilities.py::TestChangeRequestGraphQLCapabilities::test_admin_has_create_change_request_permission']
missing public api export
============================= test session starts ==============================
platform linux -- Python 3.14.3, pytest-9.0.3, pluggy-1.6.0
django: version: 6.0.4, settings: tests.pytest_settings (from ini)
rootdir: /workspaces/knowledge_hub/knowledgehub_generalmanager
configfile: backend/pytest.ini
plugins: cov-7.1.0, Faker-40.13.0, django-4.12.0, anyio-4.13.0
collected 1 item

backend/apps/project_management/tests/integration/test_change_request_graphql_capabilities.py F

=================================== FAILURES ===================================
_ TestChangeRequestGraphQLCapabilities.test_admin_has_create_change_request_permission _

self = <backend.apps.project_management.tests.integration.test_change_request_graphql_capabilities.TestChangeRequestGraphQLCapabilities testMethod=test_admin_has_create_change_request_permission>

    def setUp(self) -> None:
        self.admin_user = User.Factory.create(
            is_active=True,
            is_staff=True,
        )
        self.user = User.Factory.create(
            is_active=True,
            is_staff=False,
        )
        self.customer = Customer.Factory.create(
            key_account=self.admin_user,
            number=1,
        )
        self.project_phase_type = ProjectPhaseType.Factory.create()
        self.project_type = ProjectType.Factory.create()
        self.currency = Currency.Factory.create()
>       self.project = cast(Any, Project).Factory.create()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

/home/vscode/.local/lib/python3.14/site-packages/factory/base.py:539: in create
    return cls._generate(enums.CREATE_STRATEGY, kwargs)
/home/vscode/.local/lib/python3.14/site-packages/general_manager/factory/auto_factory.py:154: in _generate
    params[field.name] = get_field_value(field)
/home/vscode/.local/lib/python3.14/site-packages/general_manager/factory/factories.py:193: in get_field_value
    related_model = get_related_model(field)

field = <django.db.models.fields.related.OneToOneField: project_number>

    def get_related_model(field):
        related_model = field.related_model
        if related_model is None:
            raise MissingRelatedModelError(field.name)
        if related_model == "self":
            related_model = field.model
        if not isinstance(related_model, type) or not issubclass(
            related_model, models.Model
        ):
>           raise InvalidRelatedModelTypeError(field.name, related_model)
E           general_manager.factory.factories.InvalidRelatedModelTypeError: Related model for project_number must be a Django model class, got 'financials.AccountNumber'.

/home/vscode/.local/lib/python3.14/site-packages/general_manager/factory/factories.py:257: InvalidRelatedModelTypeError
```

### Additional context, screenshots, or references

The failing field is nullable and optional:

```python
project_number = models.OneToOneField(
    "financials.AccountNumber",
    on_delete=models.CASCADE,
    null=True,
    blank=True,
    default=None,
    related_name="project_number",
)
```

The same problem can also occur for required relations if the factory attempts to auto-create the related instance before Django has resolved the string reference, for example:

```text
Related model for project_phase_type must be a Django model class, got 'project.ProjectPhaseType'.
```

Explicitly passing all related fields avoids the issue, but that defeats the expected factory behavior for `Manager.Factory.create()`.

--- a/src/general_manager/factory/factories.py
+++ b/src/general_manager/factory/factories.py
@@ -7,6 +7,7 @@ from typing import Any, cast
 from factory.declarations import LazyFunction
 from factory.faker import Faker
 import exrex  # type: ignore[import-untyped]
+from django.apps import apps
 from django.core.validators import RegexValidator
 from django.db import models
 from datetime import date, datetime, timezone, timedelta
@@ -18,6 +19,7 @@ from general_manager.manager.general_manager import GeneralManager
 
 
 _RNG = SystemRandom()
+_NO_DEFAULT = object()
 
 
 class MissingFactoryOrInstancesError(ValueError):
@@ -96,6 +98,13 @@ def get_field_value(
         MissingRelatedModelError: When a relational field does not declare a related model.
         InvalidRelatedModelTypeError: When a relational field's related value is not a Django model class.
     """
+    if (
+        getattr(field, "null", False)
+        and getattr(field, "default", _NO_DEFAULT) is None
+        and getattr(field, "is_relation", False)
+    ):
+        return None
+
     if field.null:
         if _RNG.choice([True] + 9 * [False]):
             return None
@@ -246,16 +255,39 @@ def get_related_model(
         MissingRelatedModelError: If the field does not declare a related model.
         InvalidRelatedModelTypeError: If the resolved related model is not a Django model class.
     """
-    related_model = field.related_model
+    related_model: object = field.related_model
     if related_model is None:
         raise MissingRelatedModelError(field.name)
-    if related_model == "self":
-        related_model = field.model
+    if isinstance(related_model, str):
+        related_model = _resolve_related_model_string(field, related_model)
     if not isinstance(related_model, type) or not issubclass(
         related_model, models.Model
     ):
         raise InvalidRelatedModelTypeError(field.name, related_model)
     return cast(type[models.Model], related_model)
+
+
+def _resolve_related_model_string(
+    field: models.ForeignObjectRel | models.Field[Any, Any],
+    related_model: str,
+) -> object:
+    if related_model == "self":
+        return field.model
+    app_label: str | None
+    if "." in related_model:
+        app_label, model_name = related_model.split(".", 1)
+    else:
+        model = getattr(field, "model", None)
+        opts = getattr(model, "_meta", None)
+        app_label_value = getattr(opts, "app_label", None)
+        app_label = app_label_value if isinstance(app_label_value, str) else None
+        model_name = related_model
+    if not app_label:
+        return related_model
+    try:
+        return apps.get_model(app_label, model_name)
+    except LookupError:
+        return related_model
 
 
 def get_many_to_many_field_value(

--- a/tests/unit/test_factories.py
+++ b/tests/unit/test_factories.py
@@ -431,6 +431,48 @@ class TestGetManyToManyFieldValue(TestCase):
         self.assertIn("test_field", str(ctx.exception))
         self.assertIn("must be a Django model class", str(ctx.exception))
 
+    def test_get_related_model_resolves_dotted_string_reference(self):
+        """String relation references should resolve through Django's app registry."""
+        from general_manager.factory.factories import get_related_model
+
+        field = Mock()
+        field.related_model = "financials.AccountNumber"
+        field.name = "project_number"
+
+        with patch(
+            "django.apps.apps.get_model", return_value=DummyForeignKey
+        ) as get_model:
+            self.assertIs(get_related_model(field), DummyForeignKey)
+
+        get_model.assert_called_once_with("financials", "AccountNumber")
+
+    def test_nullable_relation_with_default_none_returns_none_before_resolution(self):
+        """Nullable relation fields with default=None should not require resolving a related model."""
+        from general_manager.factory.factories import get_field_value
+
+        field = Mock()
+        field.related_model = "financials.AccountNumber"
+        field.name = "project_number"
+        field.null = True
+        field.default = None
+
+        def custom_isinstance(obj, cls):
+            if cls == models.OneToOneField:
+                return True
+            return isinstance(obj, cls)
+
+        with (
+            patch("general_manager.factory.factories._RNG.choice", return_value=False),
+            patch(
+                "general_manager.factory.factories.isinstance",
+                side_effect=custom_isinstance,
+            ),
+            patch("django.apps.apps.get_model") as get_model,
+        ):
+            self.assertIsNone(get_field_value(field))
+
+        get_model.assert_not_called()
+
     def test_nullable_foreign_key_without_factory_returns_none(self):
         """Nullable foreign keys without factories or instances should fall back to None."""
         from general_manager.factory.factories import get_field_value

